src/sage/schemes/curves/zariski_vankampen.py

```diff
@@ -1699,23 +1886,32 @@ def fundamental_group_arrangement(flist, simplified=True, projective=False, puis
         R = PolynomialRing(QQ, ('x', 'y'))
         f = R(1)
     x, y = R.gens()
-    F = R.base_ring()
-    flist1 = list(flist)
-    d = f.degree(y)
-    while not f.coefficient(y**d) in F:
-        flist1 = [g.subs({x: x + y}) for g in flist1]
-        f = prod(flist1)
-        d = f.degree(y)
-    if projective:
-        while f.degree(y) < f.degree():
-            flist1 = [g.subs({x: x + y}) for g in flist]
-            f = prod(flist1)
-    if not flist1:
+    flist1 = tuple(list(flist).copy())
```

Comment:
```suggestion
    flist1 = tuple(flist)
```
No need to copy stuff (unless you meant a deep copy?).